kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -109,9 +162,49 @@ private void addHandlers(SocketChannel ch, VirtualCluster virtualCluster) {
             pipeline.addLast(new KafkaAuthnHandler(ch, authnHandlers));
         }
 
-        var netFilter = new UpstreamBrokerAddressCachingNetFilter(config, virtualCluster);
+        var frontendHandler = new KafkaProxyFrontendHandler(context -> {
+            var filterChainFactory = new FilterChainFactory(config, virtualCluster.getClusterEndpointProvider());
```

Comment:
can we pass `virtualCluster` instead and delete the `getClusterEndpointProvider` method?